lib/galaxy/files/sources/_rdm.py

```diff
@@ -199,7 +199,7 @@ def get_authorization_token(self, user_context: OptionalUserContext) -> str:
             effective_props = self._serialization_props(user_context)
             token = effective_props.get("token")
         if not token:
-            raise AuthenticationRequired(
+            raise MessageException(
```

Comment:
`AuthenticationRequired` seems fine here, since it's a subclass of MessageException

Reply:
This is then translated to `Authentication required for this request`, which seems fine?

Reply:
I guess unintentionally, yes 😆. I do think the 403 that comes with `AuthenticationRequired ` is better than the 400 for `MessageException` but we could certainly add another exception class if we want to be even more precise.

Reply:
Lol, yes, but it will be replaced anyway with `Please provide a personal access token in your user's preferences for '{self.label}'`

I will use AuthenticationRequired in those cases then :+1: